github_connector/tests/test_github_analysis_rule.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (C) 2020: Odoo Community Association (OCA)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+import logging
+
+from odoo.tests.common import SavepointCase
+
+_logger = logging.getLogger(__name__)
+
+
+class TestGithubAnalysisRule(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestGithubAnalysisRule, cls).setUpClass()
+
+        cls.analysis_rule_group_misc = cls.env["github.analysis.rule.group"].create(
+            {"name": "Misc"}
+        )
+        cls.analysis_rule_python = cls.env["github.analysis.rule"].create(
+            {
+                "name": "Python files",
+                "group_id": cls.analysis_rule_group_misc.id,
+                "paths": "/*/*.py",
```

Comment:
why don't you reuse demo data, you just added ?

self.env.ref("github_analysis_rule_1_demo") should work

Reply:
@legalsylvain no, we never use demo data for tests. That's a very bad practice and Odoo itself has finally corrected this in v14 (mostly). There are tons of reasons for that that require an specific topic for that.

Reply:
if you have a link with the reasons explained, I'm interested ! I never faced issues regarding that topic.

but no problem. LGTM

Reply:
I don't have right now the explanations given in other PRs or upstream. but a brief summary is:

- They are not re-entrant. That demo data may be modified in the DB where you are executing the tests.
- Not only that, even not being modified manually, can be modified by other dependent modules. It's happening for example with the standard list price in core addons that forces us to make subsequent fixes when integrating modules: https://github.com/OCA/e-commerce/pull/403/files#diff-7b8a8f3a73e2fc7cad9f76765050b44bR24
- They spread data across several files, not being easily to predict the final result.

The cons is to include more code, but it's not very hard and it's easily done once you get used to, all in the setUp method.